inst/include/dust/filter.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef DUST_FILTER_HPP
+#define DUST_FILTER_HPP
+
+namespace dust {
+
+template <typename T>
+std::vector<typename T::real_t> filter(T * obj,
```

Comment:
Would `Dust<T>` be a better template?

Reply:
Or does that make getting real_t annoying. Just thinking that obj feels a bit too general

Reply:
hmm, good shout, will check